src/flexmeasures_client/client.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import asyncio
+import socket
+from dataclasses import dataclass
+from typing import Any, cast
+
+import async_timeout
+import pandas as pd
+from aiohttp.client import ClientError, ClientSession
+from yarl import URL
+
+
+@dataclass
+class FlexmeasuresClient:
+    """Main class for connecting to the Flexmeasures API"""
+
+    password: str
+    email: str
+    access_token: str = None
+    scheme: str = "http"
+    host: str = "localhost:5000"
+    local: bool = False
+    version: str = "/v3_0/"
+
+    request_timeout: float = 10.0
+    session: ClientSession | None = None
+
```

Comment:
Maybe a dedicated `__init__` function, where one can start a client with chosen parametrisation, e.g. for `host`, and we get an error if email and password are not given?

Not sure what the best way is to store credentials like password, we could figure that out in a follow-up issue.